docs/authentication/3id-did/security.md

```diff
@@ -0,0 +1,13 @@
+# Security
+
+The 3ID Connect system allows blockchain accounts to be used as an authentication method to the keys used for the users 3ID. In order to authenticate a user signs a message with their wallet. The entropy is used to generate a [Key DID](./authentication/key-did/method.md) which in turn is used to decrypt the seed used for the 3ID.
+
+The main security concern here is that the seed used for the 3ID is stored within the 3ID iframe. While this restricts the app using 3ID Connect from doing whatever it wants with the seed, a malicious browser plugin could easily pick up the seed from the web browser and take full control over the users 3ID. While are multiple paths that can be used to mitigate this issue, it's currently considered to be a fair tradeoff to improve UX for users. In contrast to a users crypto wallet, the 3ID doesn't actually hold or control any cryptocurrency; it only controls data read/writes. This means that the incentive for a hacker to execute an attack is much smaller since there is no immediate financial reward. Also note that this security issue is not inherent to 3ID itself, it's just related to how 3ID is used within 3ID connect.
+
+## Possible security improvements
+
+### External 3ID controller
+The 3ID DID document has different public keys used for writing and controlling the DID document itself. Currently however, these keys are generated from the same seed. This means that if someone gets access to this seed and rotate keys using the controller key they will own the 3ID forever. One possible mitigation strategy is to store the controller key in a more secure manner. This would make it possible to recover if the day to day writing public keys are compromised. An example solution here could be to store the controller key in a MetaMask Snap.
```

Comment:
```suggestion
The 3ID DID document has different public keys used for writing and controlling the DID document itself. Currently however, these keys are generated from the same seed. This means that if someone gets access to this seed and rotates keys using the controller key they will own the 3ID forever. One possible mitigation strategy is to store the controller key in a more secure manner. This would make it possible to recover if the day to day writing public keys are compromised. An example solution here could be to store the controller key in a MetaMask Snap.
```

Reply:
wait, are you saying that if the day-to-day writing keys are compromised then the attacker can also compromise the seed and get the controller key?  I would have thought the transformation from seed to keys was one-way.

Reply:
Transformation is one way, but we store the seed in browser storage. Regardless only way around this is to have the controller key never touch the browser window!

Reply:
I understand that storing the seed in the browser is insecure and risky, I'm trying to understand some of the wording of this paragraph that implies a relationship between losing the public writing/signing keys and the seed.

> The 3ID DID document has different public keys used for writing and controlling the DID document itself. Currently however, these keys are generated from the same seed. **This means that** [...]

"this means that" implies that the fact that the writing and controller keys come from the same seed makes the situation worse if the seed is lost, but if the seed was used just for the controller key wouldn't it be the same situation?

>  This would make it possible to recover if the day to day writing public keys are compromised

This implies to me that if the day-to-day writing keys are compromised that also compromises the seed/controller key.

Reply:
"This would make it possible to recover from a scenario where the day-to-day writing public keys are compromised because the controller key would still be in control of the user"

How is that different than today?  It seems like this would only not be the case today if losing the public signing keys also results in losing the controller key, but you just said that isn't the case, so I'm confused.

Reply:
Unresolved this thread since I do want to make sure I understand this.  Your updated wording in the newest version of the PR makes me think that losing the day-to-day signing keys *does* in fact grant you access to the controller key as well, since they have the same seed.  Is that right?

Reply:
No that's not right. However, since you store the seed rather than the keys themselves in the iframe both of the keys can be derived.

Reply:
sigh I'm still not following then.  You keep saying two things that seem incompatible to me.  You say that losing the signing keys doesn't compromise the controller key, but then you also say: "This would make it possible to recover from a scenario where the day-to-day writing public keys are compromised because the controller key would still be in control of the user" <- That implies to me that today, that isn't the case.  In other words, today, if the day-to-day writing public keys are compromised then the controller key would *not* still be in control of the user.

Reply:
I feel like I've said this multiple times now. We store the **seed** which generates the day-to-day key as well as the controller key in the browser. If this **seed** is compromised then users fully loose control.

The suggested change here is to use a separate seed for the controller key that doesn't ever touch the browser.

Reply:
yes I agree that you have said that multiple times now 😛 The problem is I still don't think that what you are saying here in the comments matches what is written in the PR.

Maybe we should try to discuss live since we seem to be talking past each other in this comment thread.  Perhaps right after social hour today?